NitroxPatcher/Patches/Persistent/GameSettings_SerializeInputSettings_Patch.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Reflection;
+using HarmonyLib;
+using NitroxClient.MonoBehaviours.Gui.Input;
+using NitroxClient.MonoBehaviours.Gui.Input.KeyBindings;
+using NitroxClient.Serialization;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Persistent
+{
+    public class GameSettings_SerializeInputSettings_Patch : NitroxPatch, IPersistentPatch
+    {
+        private static readonly MethodInfo TARGET_METHOD = Reflect.Method(() => GameSettings.SerializeInputSettings(default(GameSettings.ISerializer)));
+
+        public static void Postfix(GameSettings.ISerializer serializer)
+        {
+            ClientConfig cfg = ClientConfig.Load(NitroxUser.AppDataPath);
+            KeyBindingManager keyBindingManager = new();
+            string serializerFormat = "Input/Binding/{0}/{1}/{2}";
+
+            foreach (GameInput.BindingSet bindingSet in Enum.GetValues(typeof(GameInput.BindingSet)))
+            {
+                foreach (KeyBinding keyBinding in keyBindingManager.KeyboardKeyBindings)
+                {
+                    Log.Debug($"Getting keybinding: {keyBinding.Device}, {keyBinding.Label} ({keyBinding.Button}), {bindingSet}");
+                    string binding = GameInput.GetBinding(keyBinding.Device, keyBinding.Button, bindingSet);
+
+                    // We need to assign the correct binding for primary and secondary binding sets to the relevant area of the config.
+                    switch ((KeyBindingValues)keyBinding.Button)
+                    {
+                        case KeyBindingValues.CHAT when bindingSet == GameInput.BindingSet.Primary:
+                            cfg.OpenChatKeybindPrimary = binding;
+                            break;
+                        case KeyBindingValues.FOCUS_DISCORD when bindingSet == GameInput.BindingSet.Primary:
+                            cfg.FocusDiscordKeybindPrimary = binding;
+                            break;
+                        case KeyBindingValues.CHAT when bindingSet == GameInput.BindingSet.Secondary:
+                            cfg.OpenChatKeybindSecondary = binding;
+                            break;
+                        case KeyBindingValues.FOCUS_DISCORD when bindingSet == GameInput.BindingSet.Secondary:
+                            cfg.FocusDiscordKeybindSecondary = binding;
+                            break;
+                    }
+
+                    GameInput.SetBindingInternal(keyBinding.Device, keyBinding.Button, bindingSet, binding);
```

Comment:
The purpose of this line isn't clear to me and doesn't seem to suit this patch well. Do you know if it's safe to remove this line?

Reply:
I thought (may be incorrect) that this handled setting the value of our bind in game itself, as the original method has no knowledge of our binds

This was set to serialize the input out (which stored it in options.bin) but as we no longer need that, I just used the internal method.

I can remove and test, in case it isn't needed

Reply:
Both patches (`GameInput_SetupDefaultKeyboardBindings_Patch` and `GameSettings_SerializeInputSettings_Patch`) read from `keyBindingManager.KeyboardKeyBindings`, which is our type/code. So I think it's fine to remove the line as `GameInput.SetBindingInternal` is already executed in `GameInput_SetupDefaultKeyboardBindings_Patch` and when a user edits its config it will use the game code that calls `GameInput.SetBindingInternal`.

Reply:
I'll make the change as soon as I get the chance 🙏